lib/enigma.rb

```diff
@@ -40,4 +40,21 @@ def decrypt(message, key, date)
     decryption[:date] = date
     decryption
   end
+
+  def current_date
+    now = Time.now
+    now.strftime("%m%d%y")
+  end
+
+  def generate_number
+    rand(100000).to_i
+  end
+
+  def generate_key
+    key_number = generate_number.to_s
+    until key_number.length == 5
+      key_number.prepend("0")
+    end
+    key_number
+  end
```

Comment:
tests added for these cases. This functionality is critical to the encrypt and decrypt files which will not take a date or key as an argument.